packages/astro/src/@types/astro.ts

```diff
@@ -1096,12 +1087,111 @@ export interface AstroAdapter {
 
 type Body = string;
 
-export interface APIContext {
+// Shared types between `Astro` global and API context object
+interface AstroSharedContext<Props extends Record<string, any> = Record<string, any>> {
+	/**
+	 * The address (usually IP address) of the user. Used with SSR only.
+	 */
+	clientAddress: string;
+	/**
+	 * Utility for getting and setting cookies values.
+	 */
 	cookies: AstroCookies;
-	params: Params;
+	/**
+	 * Information about the current request. This is a standard [Request](https://developer.mozilla.org/en-US/docs/Web/API/Request) object
+	 */
 	request: Request;
+	/**
+	 * A full URL object of the request URL.
+	 */
+	url: URL;
+	/**
+	 * Parameters passed to a dynamic page generated.
+	 */
+	params: Params;
+	/**
+	 * List of props passed.
+	 */
+	props: Props;
+	/**
+	 * Redirect to another page (**SSR Only**)
```

Comment:
Micro-nit 🐜 
The other comments all ended in a full-stop.

```suggestion
	 * Redirect to another page (**SSR Only**).
```